.github/workflows/tests.yml

```diff
@@ -24,3 +27,61 @@ jobs:
           && !contains(github.event.pull_request.labels.*.name, 'ci:run-full-test-suite')
           || false
          }}
+
+  upload-stable-deps:
+    needs: tests
+    runs-on: ubuntu-latest
+    if: ${{ github.event_name == 'push' }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          fetch-tags: true
+          sparse-checkout: .github
+
+      - name: Prepare local repo
+        run: |
+          git fetch
+          git config user.name "GitHub Actions Bot"
+          git config user.email "<>"
+          if git ls-remote --exit-code origin "refs/heads/${{ env.DEPS_BRANCH }}"; then
+            git checkout "${{ env.DEPS_BRANCH }}"
+            git rebase origin/master
+          else
+            git checkout master
+            git checkout -b "${{ env.DEPS_BRANCH }}"
+          fi
+          rm -f .github/stable/*.txt
+
+      - name: Download artifacts
+        uses: actions/download-artifact@v4
+        with:
+          pattern: frozen-*
+          path: .github/stable/
+          merge-multiple: true
+
+      - name: Determine if changes have been made
+        id: changed
+        run: |
+          echo "has_changes=$(git status --porcelain | wc -l | awk '{print $1}')" >> $GITHUB_OUTPUT

```

Comment:
# Question

Why do we need the `| awk '{print $1}')` part?

Reply:
just the formatting of `wc` results:
```bash
$ git status --porcelain | wc -l
       2
$ x='       2'                                      
$ y='2'                                             
$ val=`git status --porcelain | wc -l`
$ test $val = $x && echo "success"
success
$ test $val = $y && echo "success"
```

Reply:
Ah, I see! Perhaps `| tr -d ' '` would be less overkill but using `awk` is fine too!